src/aiagentservice.js

```diff
@@ -308,13 +313,28 @@ export default class AiAgentService {
                 return await this.fetchAndProcessGptResponse(prompt, parent, retries - 1);
             }
             let errorMessage;
-            switch ((error === null || error === void 0 ? void 0 : error.name) || ((_c = error === null || error === void 0 ? void 0 : error.message) === null || _c === void 0 ? void 0 : _c.trim())) {
+            const jsonMessage = this.isValidJSON(error === null || error === void 0 ? void 0 : error.message);
+            let status;
+            if (jsonMessage) {
+                const errorObj = JSON.parse(error === null || error === void 0 ? void 0 : error.message);
+                status = errorObj.status;
+            }
+            switch (status || (error === null || error === void 0 ? void 0 : error.name) || ((_c = error === null || error === void 0 ? void 0 : error.message) === null || _c === void 0 ? void 0 : _c.trim())) {
                 case 'ReadableStream not supported':
                     errorMessage = t('Browser does not support readable streams');
                     break;
                 case 'AiAgent: Fetch failed':
```

Comment:
Also, please check if this case is not preventing more precise error messages from materializing. Maybe only keep this message in the default behavior? I'm not sure I didn't do a deep analysis here but please analyze this carefully